### PR TITLE
Adds ci testing

### PR DESCRIPTION
**Pull request by mmeier93**
## This PR does the following:

It adds basic CI testing to the pipeline.

## Details:

- Added a `conf/basic_test.config` profile: runs METAL on 3 datasets.

- Added a `conf/extra_flags.config` profile: runs METAL on 3 datasets as well as 2 optional flags.

- In order for the flags to be correctly accepted:
  - Removed the space that was set when setting the params (otherwhise the flags were not "padded" to the wall like the other `METAL` options
  - Swapped around `$process_commands` and `!{extra_flags}`. Indeed in their original position, `METAL` would ignore the extra flags (i.e. as soon as it sees the `PROCESS` commands, it stopped caring about other options afterwards).

## Comments:

The files used for testing (i.e. the input `list-summary-statistics.csv`), as well as the correspond CSV files listed in it are currently on my personal S3 bucket and will need to be moved to the common bucket.

## Testing:

Successful CI testing: https://github.com/lifebit-ai/metagwas/actions/runs/421249851

Successful CloudOS job: https://cloudos.lifebit.ai/app/jobs/5fd796f1aec0ef0112b768e7

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## 1 - Quick Start
 
 ```
-nextflow run main.nf --studies testdata/list-summary-statistics.csv
+nextflow run main.nf --studies https://testdata-magda.s3-eu-west-1.amazonaws.com/metagwas/list-summary-statistics.csv
```
Comment (cgpu) [inline README.md:6]:
```Suggestion
nextflow run main.nf --studies https://lifebit-featured-datasets.s3-eu-west-1.amazonaws.com/projects/gel/metagwas/list-summary-statistics.csv
```

File: conf/flags_test.config [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * Defines bundled input files and everything required
+ * to run a fast and simple test. Use as follows:
+ * nextflow run main.nf -profile flags_test
+ */
+
+
+docker.enabled = true
+
+params  {
+    studies = "https://testdata-magda.s3-eu-west-1.amazonaws.com/metagwas/list-summary-statistics.csv"
```
Comment (cgpu) [inline conf/flags_test.config:14]:
```Suggestion
    studies = "https://lifebit-featured-datasets.s3-eu-west-1.amazonaws.com/projects/gel/metagwas/list-summary-statistics.csv"
```

File: docs/usage.md [hunk 3]
```diff
@@ -13,15 +13,13 @@ This pipeline currently assumes all the studies are formatted as SAIGE output fi
 The typical command for running the pipeline is as follows:
 
 ```
-nextflow run main.nf --studies testdata/list-summary-statistics.csv
+nextflow run main.nf --studies https://testdata-magda.s3-eu-west-1.amazonaws.com/metagwas/list-summary-statistics.csv
```
Comment (cgpu) [inline docs/usage.md:16]:
```Suggestion
nextflow run main.nf --studies https://lifebit-featured-datasets.s3-eu-west-1.amazonaws.com/projects/gel/metagwas/list-summary-statistics.csv
```

File: nextflow.config [hunk 4]
```diff
@@ -31,34 +31,34 @@ params {
 
   // 2 - METAL options for filtering input files
   
-  addfilter  = false
+  addfilter     = false
```
Comment (cgpu) [inline nextflow.config:34]: ah the fresh OCD in the morning! 😃 
Believe it or not there is an R function that is also an Rstudio addin that does this, [`remedy::align_equal()`](https://thinkr-open.github.io/remedy/#align)

I have used this more than I should ever admit.

File: main.nf [hunk 5]
```diff
@@ -86,50 +86,50 @@ extra_flags = ""
 
 // 1 - METAL options for describing input files
```
Comment (cgpu) [inline main.nf:87]: Awesome section comments, appreciate this for readability 👍

**Review by cgpu — APPROVED** [review]
Looks good to me!

File: nextflow.config (hunk 4 above)
Comment (mmeier93) [reply nextflow.config:34]: :D Thanks for tip!

File: main.nf (hunk 5 above)
Comment (mmeier93) [reply main.nf:87]: 👍

**Comment by mmeier93** [comment]
> Looks good to me! 

Thanks! 👍

**Pull request merged**